### PR TITLE
Adding some clarification around granularity of PRs. [ci skip]

### PR DESCRIPTION
**Pull request by khandpur**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ The submitted code should adhere to the following:
 
 ### Do's
 
+* Keep pull requests focused on addressing one issue or feature. If multiple separate issues are fixed by a pull request, prefer commiting the fixes under multiple commits, one for each issue.
```
Comment (tirodkar) [inline CONTRIBUTING.md]: Nit: Break this into multiple lines.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ The submitted code should adhere to the following:
...
 * For every pull request, make sure all commits are consolidated into one commit before
```
Comment (axi0mX) [inline CONTRIBUTING.md]: s/all commits are consolidated into one commit/all commits for a related issue or feature are consolidated into a single commit/

**Comment by bootstraponline** [comment]
`[ci skip]` has to go in the commit message, not the pull request title.

**Comment by khandpur** [comment]
@bootstraponline yup, i realized that after triggering the build. Manually cancelled it.

**Comment by khandpur** [comment]
PTAL

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -24,15 +24,18 @@ The submitted code should adhere to the following:
 
 ### Do's
 
-* For every pull request, make sure all commits are consolidated into one commit before
-  sending the pull request.
+* Keep pull requests focused on addressing one issue or feature. If multiple separate issues
+  are fixed by a pull request, prefer commiting the fixes under multiple commits, one for each issue.
+* For every pull request, make sure all commits for a related issue or feature are consolidated
+  into a single commit before sending the pull request.
 * Before submitting any code changes, please ensure that the [Unit Tests](https://github.com/google/EarlGrey/tree/master/Tests/UnitTests)
```
Comment (bootstraponline) [inline CONTRIBUTING.md]: it's easier to review pull requests as individual commits and then have GitHub auto squash them via the new squash button.

https://github.com/blog/2141-squash-your-commits

Comment (khandpur) [inline CONTRIBUTING.md]: I'll just remove it. We added it because Github did not have a squash commit option. Now that it does, this point is moot.

**Comment by bootstraponline** [comment]
👍

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -24,15 +24,16 @@ The submitted code should adhere to the following:
 
 ### Do's
 
-* For every pull request, make sure all commits are consolidated into one commit before
-  sending the pull request.
+* Keep pull requests focused on addressing one issue or feature. If multiple separate issues
```
Comment (bootstraponline) [inline CONTRIBUTING.md]: [Based on what happened today](https://github.com/google/EarlGrey/commit/f2cc0c464cc99d280bf75561ba44b844e83e382c), this info has to change.

>  If multiple separate issues are fixed by a pull request, prefer committing the fixes under multiple commits, one for each issue.

Multiple separate issues require unique pull requests if every pull request is going to be squashed.

Comment (tirodkar) [inline CONTRIBUTING.md]: PR's when merged will always be squashed. We shouldn't be having multiple fixes in one pull request or break larger ones into smaller, separate PR's.

> If multiple separate issues... prefer commiting the fixes under multiple commits, one for each issue

should be removed.

**Comment by khandpur** [comment]
That is because I turned off merging commits one by one. I have allowed it again. @bootstraponline @tirodkar what is so wrong with > 1 commit per pr if it fixes multiple smaller issues? Is the git history not preserved with the commits?

**Comment by bootstraponline** [comment]
> @bootstraponline @tirodkar what is so wrong with > 1 commit per pr if it fixes multiple smaller issues?

This requires the people doing the merging to have an understanding of the basic GitHub workflow which is not currently true. It's easier to keep things simple with 1 logical change = 1 PR = 1 merged commit. Otherwise a judgement call has to be made on each PR about if it's a merge or a squash. In the two situations that's happened, the wrong choice was made.

**Pull request merged**